src/ucp/rndv/proto_rndv.inl

```diff
@@ -234,32 +234,41 @@ ucp_proto_rndv_bulk_max_payload(ucp_request_t *req,
     size_t lane_offset, max_payload, scaled_length;
 
     /* Request remaining length cannot be lower than minimal payload size */
-    ucs_assertv((req_length - req_offset) >= min_frag,
-                "req=%p length-offset=(%zu-%zu)=%zu min_frag=%zu", req,
-                req_length, req_offset, req_length - req_offset, min_frag);
+    ucs_assertv(req_remainder >= min_frag,
+                "req=%p req_remainder=(%zu-%zu)=%zu min_frag=%zu", req,
+                req_length, req_offset, req_remainder, min_frag);
 
     if (ucs_likely(total_length < max_frag_sum)) {
         /* Each lane sends less than its maximal fragment size */
         scaled_length = ucp_proto_multi_scaled_length(lpriv->weight_sum,
                                                       total_length);
-        max_payload   = scaled_length - total_offset;
+
+        ucs_assertv(scaled_length >= total_offset,
+                    "req=%p scaled_length=%zu total_offset=%zu "
+                    "total_length=%zu weight_sum=%zu%% ",
+                    req, scaled_length, total_offset, total_length,
+                    ucp_proto_multi_scaled_length(lpriv->weight_sum, 100));
```

Comment:
why is 100 used here?

Reply:
since we print weight as percents (%% in the fmt string)